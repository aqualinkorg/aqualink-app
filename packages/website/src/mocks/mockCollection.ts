import { CollectionDetails } from "../store/Collection/types";
import { mockReef } from "./mockReef";
import { mockUser } from "./mockUser";

export const mockCollection: CollectionDetails = {
  id: 1,
  isPublic: false,
  name: "Mock Collection",
  user: mockUser,
  reefs: [mockReef],
  reefIds: [1],
};
