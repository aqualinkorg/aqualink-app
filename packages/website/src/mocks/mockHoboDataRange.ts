import { TimeSeriesDataRange } from "../store/Reefs/types";

export const mockHoboDataRange: TimeSeriesDataRange = {
  hobo: {
    bottomTemperature: [
      {
        minDate: "2017-11-02T14:59:08.000Z",
        maxDate: "2020-02-02T20:59:08.000Z",
      },
    ],
    alert: [],
    dhw: [],
    satelliteTemperature: [],
    topTemperature: [],
    sstAnomaly: [],
    significantWaveHeight: [],
    wavePeakPeriod: [],
    waveMeanDirection: [],
    windSpeed: [],
    windDirection: [],
  },
  spotter: {
    bottomTemperature: [],
    alert: [],
    dhw: [],
    satelliteTemperature: [],
    topTemperature: [],
    sstAnomaly: [],
    significantWaveHeight: [],
    wavePeakPeriod: [],
    waveMeanDirection: [],
    windSpeed: [],
    windDirection: [],
  },
  sofarApi: {
    bottomTemperature: [],
    alert: [],
    dhw: [],
    satelliteTemperature: [],
    topTemperature: [],
    sstAnomaly: [],
    significantWaveHeight: [],
    wavePeakPeriod: [],
    waveMeanDirection: [],
    windSpeed: [],
    windDirection: [],
  },
};
