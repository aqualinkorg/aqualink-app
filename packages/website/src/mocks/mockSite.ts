import { CollectionMetrics, Point, Site } from 'store/Sites/types';
import { mockUser } from './mockUser';

const now = new Date();
const minutesAgo = 5;
const dailyDataDate = new Date(
  now.getTime() - minutesAgo * 60000,
).toISOString();

export const mockSite: Site = {
  id: 1,
  name: '',
  polygon: {
    coordinates: [0, 0],
    type: 'Point',
  },
  hasHobo: false,
  sensorId: null,
  maxMonthlyMean: 0,
  depth: 0,
  status: 'in_review',
  videoStream: null,
  region: { name: 'Hawaii' },
  admins: [mockUser],
  surveyPoints: [],
  stream: null,
  historicalMonthlyMean: [],
  display: true,
  dailyData: [
    {
      id: 171,
      date: dailyDataDate,
      minBottomTemperature: 37,
      maxBottomTemperature: 39,
      avgBottomTemperature: 38,
      degreeHeatingDays: 34,
      topTemperature: 29,
      satelliteTemperature: 23,
      minWaveHeight: 2,
      maxWaveHeight: 4,
      avgWaveHeight: 3,
      waveMeanDirection: 205,
      waveMeanPeriod: 28,
      minWindSpeed: 3,
      maxWindSpeed: 5,
      avgWindSpeed: 4,
      windDirection: 229,
    },
  ],
  collectionData: {
    topTemperature: 9.54,
    bottomTemperature: 10.719999999999999,
    satelliteTemperature: 15.8900003433228,
    dhw: 0,
    tempWeeklyAlert: 0,
    sstAnomaly: 1.2,
  },
  reefCheckSurveys: [],
  reefCheckSites: [],
  reefCheckData: null,
};

export const generateMockSite = ({
  id,
  name,
  regionName,
  deployed,
  historicMax,
  collectionData,
  polygon,
}: {
  id: number;
  name: string;
  regionName: string;
  deployed: boolean;
  historicMax: number;
  collectionData: CollectionMetrics;
  polygon: Point;
}): Site => ({
  ...mockSite,
  id,
  name,
  region: { ...mockSite.region, name: regionName },
  polygon,
  status: deployed ? 'deployed' : mockSite.status,
  sensorId: deployed ? 'SPOT-0930' : mockSite.sensorId,
  collectionData,
  maxMonthlyMean: historicMax,
});
