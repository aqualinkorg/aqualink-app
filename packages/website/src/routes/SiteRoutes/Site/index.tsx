import React, { useEffect } from "react";
import {
  withStyles,
  WithStyles,
  createStyles,
  Container,
  Box,
} from "@material-ui/core";
import { Alert } from "@material-ui/lab";
import { useSelector, useDispatch } from "react-redux";
import { Link, RouteComponentProps } from "react-router-dom";
import classNames from "classnames";
import NotFoundPage from "../../NotFound/index";
import SiteNavBar from "../../../common/NavBar";
import SiteFooter from "../../../common/Footer";
import SiteDetails from "../../../common/SiteDetails";
import SiteInfo from "./SiteInfo";
import {
  siteDetailsSelector,
  siteRequest,
  siteTimeSeriesDataRangeRequest,
  clearTimeSeriesData,
  clearTimeSeriesDataRange,
  siteOceanSenseDataRequest,
  clearOceanSenseData,
  liveDataRequest,
  liveDataSelector,
  siteLoadingSelector,
} from "../../../store/Sites/selectedSiteSlice";
import {
  surveysRequest,
  surveyListSelector,
} from "../../../store/Survey/surveyListSlice";
import { userInfoSelector } from "../../../store/User/userSlice";
import { isAdmin } from "../../../helpers/user";
import { findAdministeredSite } from "../../../helpers/findAdministeredSite";
import { User } from "../../../store/User/types";
import { localizedEndOfDay } from "../../../common/Chart/MultipleSensorsCharts/helpers";
import { sortByDate, subtractFromDate } from "../../../helpers/dates";
import { oceanSenseConfig } from "../../../constants/oceanSenseConfig";
import { useQueryParams } from "../../../hooks/useQueryParams";
import { findSurveyPointFromList } from "../../../helpers/siteUtils";
import LoadingSkeleton from "../../../common/LoadingSkeleton";
import { Site as SiteType } from "../../../store/Sites/types";

const getAlertMessage = (
  user: User | null,
  siteId: string,
  hasDailyData: boolean
) => {
  const userSite = findAdministeredSite(user, parseInt(siteId, 10));
  const { applied, status } = userSite || {};
  const isSiteAdmin = isAdmin(user, parseInt(siteId, 10));

  const defaultMessage =
    "Currently no Smart Buoy deployed at this site location. Real-time values are derived from a combination of NOAA satellite readings and weather models.";

  switch (true) {
    case !hasDailyData:
      return "Welcome to your virtual site, data is loading, please come back in a few hours. This site will be visible publicly as soon as it has been approved by the Aqualink team.";

    case !isSiteAdmin:
      return defaultMessage;
    case !applied:
      return (
        <div>
          {defaultMessage} Apply for an Aqualink Smart Buoy
          <span> </span> <Link to={`/sites/${siteId}/apply`}>here</Link>.
        </div>
      );

    case status === "in_review":
      return (
        <div>
          {defaultMessage} Your application for an Aqualink Smart Buoy is being
          reviewed. You can check your application<span> </span>
          <Link to={`/sites/${siteId}/apply`}>here</Link>.
        </div>
      );

    case status === "approved":
      return "Your application for an Aqualink Smart Buoy has been approved.";

    case status === "shipped":
      return "Your Smart Buoy is on its way! Mark it as 'deployed' once it's installed to access its data.";

    case status === "rejected":
      return (
        <div>
          Your application for an Aqualink Smart Buoy was not approved at this
          time. For more information, you can contact<span> </span>
          <a href="mailto:info@aqualink.org">info@aqualink.org</a>
        </div>
      );

    default:
      return defaultMessage;
  }
};

const Site = ({ match, classes }: SiteProps) => {
  const siteDetails = useSelector(siteDetailsSelector);
  const siteLoading = useSelector(siteLoadingSelector);
  const user = useSelector(userInfoSelector);
  const surveyList = useSelector(surveyListSelector);
  const liveData = useSelector(liveDataSelector);
  const dispatch = useDispatch();
  const getQueryParam = useQueryParams();
  const siteId = match.params.id;
  const { id, dailyData, surveyPoints, timezone } = siteDetails || {};
  const querySurveyPointId = getQueryParam("surveyPoint");
  const refresh = getQueryParam("refresh");
  const { id: selectedSurveyPointId } =
    findSurveyPointFromList(querySurveyPointId, surveyPoints) || {};

  const featuredMedia = sortByDate(surveyList, "diveDate", "desc").find(
    (survey) =>
      survey.featuredSurveyMedia && survey.featuredSurveyMedia.type === "image"
  );

  const {
    id: featuredSurveyId,
    featuredSurveyMedia,
    diveDate,
  } = featuredMedia || {};
  const { surveyPoint: featuredSurveyPoint, url } = featuredSurveyMedia || {};

  const hasSpotterData = Boolean(liveData?.topTemperature);

  const hasDailyData = Boolean(dailyData && dailyData.length > 0);

  const today = localizedEndOfDay(undefined, timezone);

  const siteWithFeaturedImage: SiteType | undefined = siteDetails
    ? { ...siteDetails, featuredImage: url }
    : undefined;

  const isLoading = !siteWithFeaturedImage;

  useEffect(() => {
    if (refresh === "true") {
      dispatch(clearTimeSeriesDataRange());
      dispatch(clearTimeSeriesData());
      dispatch(clearOceanSenseData());

      dispatch(siteRequest(siteId));
      dispatch(liveDataRequest(siteId));
      dispatch(surveysRequest(siteId));
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [refresh]);

  // Fetch site and surveys
  useEffect(() => {
    dispatch(siteRequest(siteId));
    dispatch(liveDataRequest(siteId));
    dispatch(surveysRequest(siteId));

    return () => {
      dispatch(clearTimeSeriesDataRange());
      dispatch(clearTimeSeriesData());
      dispatch(clearOceanSenseData());
    };
  }, [dispatch, siteId]);

  // Fetch time series data range for the site's closest survey point
  // once the survey points are successfully fetched
  useEffect(() => {
    if (siteId === id?.toString()) {
      dispatch(
        siteTimeSeriesDataRangeRequest({
          siteId,
          pointId: selectedSurveyPointId,
        })
      );
    }
  }, [dispatch, id, selectedSurveyPointId, siteId]);

  useEffect(() => {
    if (id && oceanSenseConfig?.[id] && siteId === id.toString()) {
      dispatch(
        siteOceanSenseDataRequest({
          sensorID: "oceansense-2",
          startDate: subtractFromDate(today, "month", 6),
          endDate: today,
          latest: true,
        })
      );
    }
  }, [dispatch, id, siteId, today]);

  return (
    <>
      <SiteNavBar searchLocation />
      {!siteLoading && !siteDetails ? (
        <NotFoundPage />
      ) : (
        <Container
          className={classNames({ [classes.noDataWrapper]: !hasDailyData })}
        >
          <Box marginTop="2rem">
            <LoadingSkeleton loading={isLoading} variant="text" lines={3}>
              {siteDetails && (
                <SiteInfo
                  hasDailyData={hasDailyData}
                  site={siteDetails}
                  lastSurvey={surveyList[surveyList.length - 1]?.diveDate}
                  isAdmin={isAdmin(user, parseInt(siteId, 10))}
                />
              )}
            </LoadingSkeleton>
          </Box>
          {/* Only show alert message when data is loading */}
          {!hasSpotterData && !isLoading && !hasDailyData && (
            <Box mt="1.3rem">
              <Alert severity="info">
                {getAlertMessage(user, siteId, hasDailyData)}
              </Alert>
            </Box>
          )}
          <SiteDetails
            site={siteWithFeaturedImage}
            selectedSurveyPointId={selectedSurveyPointId}
            featuredSurveyId={featuredSurveyId}
            hasDailyData={hasDailyData}
            surveys={surveyList}
            featuredSurveyPoint={featuredSurveyPoint}
            surveyDiveDate={diveDate}
          />
        </Container>
      )}
      <SiteFooter />
    </>
  );
};

const styles = () =>
  createStyles({
    noData: {
      display: "flex",
      alignItems: "center",
      height: "80vh",
    },
    noDataWrapper: {
      height: "100%",
    },
  });

interface MatchProps extends RouteComponentProps<{ id: string }> {}

type SiteProps = WithStyles<typeof styles> & MatchProps;

export default withStyles(styles)(Site);
