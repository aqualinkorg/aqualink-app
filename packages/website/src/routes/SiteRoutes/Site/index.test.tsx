/* eslint-disable fp/no-mutation */
import React from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";
import Site from ".";
import { mockSite } from "../../../mocks/mockSite";
import { mockUser } from "../../../mocks/mockUser";
import { mockSurvey } from "../../../mocks/mockSurvey";
import { mockCollection } from "../../../mocks/mockCollection";
import { mockHoboDataRange } from "../../../mocks/mockHoboDataRange";

const mockStore = configureStore([]);

window.scrollTo = jest.fn();

jest.mock("../../../common/SiteDetails/Map", () => "Mock-Map");
jest.mock(
  "../../../common/SiteDetails/FeaturedMedia",
  () => "Mock-FeaturedMedia"
);

jest.mock(
  "../../../common/Chart/MultipleSensorsCharts",
  () => "Mock-MultipleSensorsCharts"
);

jest.mock("react-chartjs-2", () => ({
  Line: () => "Mock-Line",
  Chart: {
    pluginService: {
      register: jest.fn(),
    },
  },
}));

describe("Site Detail Page", () => {
  let elementEmpty: HTMLElement;
  let elementFull: HTMLElement;
  beforeEach(() => {
    const emptyStore = mockStore({
      selectedSite: {
        details: { ...mockSite, dailyData: [] },
        timeSeriesDataRange: mockHoboDataRange,
        loading: false,
        error: null,
      },
      user: {
        userInfo: mockUser,
        error: null,
        loading: false,
      },
      homepage: {
        siteOnMap: mockSite,
      },
      sitesList: {
        list: [],
        loading: false,
        error: null,
      },
      surveyList: {
        list: [],
        loading: false,
        error: null,
      },
      survey: {
        selectedSurvey: {
          details: null,
        },
      },
      collection: {
        details: mockCollection,
        loading: false,
        error: null,
      },
    });

    const fullStore = mockStore({
      selectedSite: {
        details: mockSite,
        timeSeriesDataRange: mockHoboDataRange,
        loading: false,
        error: null,
      },
      user: {
        userInfo: mockUser,
        error: null,
        loading: false,
      },
      homepage: {
        siteOnMap: mockSite,
      },
      sitesList: {
        list: [mockSite],
        loading: false,
        error: null,
      },
      surveyList: {
        list: [mockSurvey],
        loading: false,
        error: null,
      },
      survey: {
        selectedSurvey: {
          details: mockSurvey,
        },
      },
      collection: {
        details: mockCollection,
        loading: false,
        error: null,
      },
    });

    emptyStore.dispatch = jest.fn();
    fullStore.dispatch = jest.fn();

    const mockMatch = {
      isExact: true,
      params: {
        id: "1",
      },
      path: "/sites/:id",
      url: "/sites/1",
    };

    elementEmpty = render(
      <Provider store={emptyStore}>
        <Router>
          <Site match={mockMatch} location={{} as any} history={{} as any} />
        </Router>
      </Provider>
    ).container;

    elementFull = render(
      <Provider store={fullStore}>
        <Router>
          <Site match={mockMatch} location={{} as any} history={{} as any} />
        </Router>
      </Provider>
    ).container;
  });

  it("should render with given state from Redux store", () => {
    expect(elementEmpty).toMatchSnapshot("snapshot-with-no-data");
  });

  it("should render with given state from Redux store", () => {
    expect(elementFull).toMatchSnapshot("snapshot-with-data");
  });
});
