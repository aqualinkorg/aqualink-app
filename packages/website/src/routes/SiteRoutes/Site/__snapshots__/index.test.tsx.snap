// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site Detail Page should render with given state from Redux store: snapshot-with-data 1`] = `
<div>
  <mock-appbar
    classname="NavBar-appBar-40"
    color="primary"
    position="static"
  >
    <mock-toolbar
      classname="NavBar-toolbar-43"
    >
      <mock-drawer
        anchor="left"
        classes="[object Object]"
        open="false"
      >
        <mock-iconbutton
          style="align-self: flex-end; margin-right: 5px; margin-top: 5px; color: white;"
        >
          <clear />
        </mock-iconbutton>
        <mock-button
          classname="MenuDrawer-menuDrawerButton-53"
          component="[object Object]"
          to="/"
        >
          <mock-typography
            variant="h6"
          >
            HOME
          </mock-typography>
        </mock-button>
        <mock-button
          classname="MenuDrawer-menuDrawerButton-53"
          component="[object Object]"
          to="/map"
        >
          <mock-typography
            variant="h6"
          >
            MAP
          </mock-typography>
        </mock-button>
        <mock-button
          classname="MenuDrawer-menuDrawerButton-53"
          component="[object Object]"
          to="/buoy"
        >
          <mock-typography
            variant="h6"
          >
            BUOY
          </mock-typography>
        </mock-button>
        <mock-button
          classname="MenuDrawer-menuDrawerButton-53"
          component="[object Object]"
          to="/drones"
        >
          <mock-typography
            variant="h6"
          >
            DRONE
          </mock-typography>
        </mock-button>
        <mock-button
          classname="MenuDrawer-menuDrawerButton-53"
          component="[object Object]"
          to="/about"
        >
          <mock-typography
            variant="h6"
          >
            ABOUT
          </mock-typography>
        </mock-button>
        <mock-button
          classname="MenuDrawer-menuDrawerButton-53"
          component="[object Object]"
          to="/faq"
        >
          <mock-typography
            variant="h6"
          >
            FAQ
          </mock-typography>
        </mock-button>
        <mock-button
          classname="MenuDrawer-menuDrawerButton-53"
          component="[object Object]"
          to="/tracker"
        >
          <mock-typography
            variant="h6"
          >
            TRACK A HEATWAVE
          </mock-typography>
        </mock-button>
        <mock-button
          classname="MenuDrawer-menuDrawerButton-53"
          component="[object Object]"
          to="/register"
        >
          <mock-typography
            variant="h6"
          >
            REGISTER A SITE
          </mock-typography>
        </mock-button>
        <mock-box
          margintop="auto"
          padding="25px"
        >
          <mock-typography
            variant="subtitle1"
          >
            Aqualink's source code is licenced under MIT. Contribute!
          </mock-typography>
          <div
            class="MuiButtonGroup-root MuiButtonGroup-contained"
            role="group"
          >
            <mock-button
              classname="MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained MenuDrawer-contributeButton-52"
              color="default"
              disabled="false"
              disableelevation="false"
              disablefocusripple="false"
              disableripple="false"
              fullwidth="false"
              href="https://github.com/aqualinkorg/aqualink-app"
              size="medium"
              starticon="[object Object]"
              target="_blank"
              variant="contained"
            >
              GitHub
            </mock-button>
            <mock-button
              classname="MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained MenuDrawer-contributeButton-52"
              color="default"
              disabled="false"
              disableelevation="false"
              disablefocusripple="false"
              disableripple="false"
              fullwidth="false"
              href="https://ovio.org/project/aqualinkorg/aqualink-app"
              size="medium"
              target="_blank"
              variant="contained"
            >
              <img
                alt="Ovio Logo"
                src="ovio_logo.png"
              />
            </mock-button>
          </div>
        </mock-box>
      </mock-drawer>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 MuiGrid-grid-sm-2 MuiGrid-grid-md-4"
        >
          <mock-box
            alignitems="center"
            display="flex"
            flexwrap="nowrap"
          >
            <mock-iconbutton
              color="inherit"
              edge="start"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </mock-iconbutton>
            <mock-link
              classname="NavBar-navBarLink-41"
              href="/map"
            >
              <mock-typography
                color="textPrimary"
                variant="h4"
              >
                Aqua
              </mock-typography>
              <mock-typography
                style="color: rgb(138, 198, 222);"
                variant="h4"
              >
                link
              </mock-typography>
            </mock-link>
          </mock-box>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-7 MuiGrid-grid-sm-4 MuiGrid-grid-md-8"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <mock-button>
              SIGN IN
            </mock-button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <mock-button>
              SIGN UP
            </mock-button>
          </div>
        </div>
      </div>
    </mock-toolbar>
  </mock-appbar>
  <mock-dialog
    maxwidth="xs"
    open="false"
    scroll="body"
  >
    <mock-card>
      <div
        class="MuiCardHeader-root RegisterDialog-dialogHeader-55"
      >
        <div
          class="MuiCardHeader-content"
        >
          <mock-typography
            classname="MuiCardHeader-title"
            component="span"
            display="block"
            variant="h5"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <div
                  class="MuiGrid-root MuiGrid-container"
                >
                  <mock-typography
                    variant="h4"
                  >
                    Aqua
                  </mock-typography>
                  <mock-typography
                    classname="RegisterDialog-dialogHeaderSecondPart-56"
                    variant="h4"
                  >
                    link
                  </mock-typography>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <mock-iconbutton
                  classname="RegisterDialog-closeButton-54"
                  size="small"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </mock-iconbutton>
              </div>
            </div>
          </mock-typography>
        </div>
      </div>
      <mock-cardcontent>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
        >
          <div
            class="MuiGrid-root RegisterDialog-dialogContentTitle-57 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-10"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <mock-typography
                color="textSecondary"
                variant="h5"
              >
                Create an account
              </mock-typography>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-10"
          >
            <form
              class="RegisterDialog-form-58"
            >
              <div
                class="MuiGrid-root RegisterDialog-textFieldWrapper-59 MuiGrid-item MuiGrid-grid-xs-12"
              >
                <mock-textfield
                  error="false"
                  fullwidth="true"
                  helpertext=""
                  id="firstName"
                  inputprops="[object Object]"
                  label="First Name"
                  name="firstName"
                  placeholder="First Name"
                  variant="outlined"
                />
              </div>
              <div
                class="MuiGrid-root RegisterDialog-textFieldWrapper-59 MuiGrid-item MuiGrid-grid-xs-12"
              >
                <mock-textfield
                  error="false"
                  fullwidth="true"
                  helpertext=""
                  id="lastName"
                  inputprops="[object Object]"
                  label="Last Name"
                  name="lastName"
                  placeholder="Last Name"
                  variant="outlined"
                />
              </div>
              <div
                class="MuiGrid-root RegisterDialog-textFieldWrapper-59 MuiGrid-item MuiGrid-grid-xs-12"
              >
                <mock-textfield
                  error="false"
                  fullwidth="true"
                  helpertext=""
                  id="organization"
                  inputprops="[object Object]"
                  label="Organization"
                  name="organization"
                  placeholder="Organization"
                  variant="outlined"
                />
              </div>
              <div
                class="MuiGrid-root RegisterDialog-textFieldWrapper-59 MuiGrid-item MuiGrid-grid-xs-12"
              >
                <mock-textfield
                  error="false"
                  fullwidth="true"
                  helpertext=""
                  id="emailAddress"
                  inputprops="[object Object]"
                  label="Email Address"
                  name="emailAddress"
                  placeholder="Email Address"
                  variant="outlined"
                />
              </div>
              <div
                class="MuiGrid-root RegisterDialog-textFieldWrapper-59 MuiGrid-item MuiGrid-grid-xs-12"
              >
                <mock-textfield
                  error="false"
                  fullwidth="true"
                  helpertext=""
                  id="password"
                  inputprops="[object Object]"
                  label="Password"
                  name="password"
                  placeholder="Password"
                  type="password"
                  variant="outlined"
                />
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
                >
                  <mock-checkbox
                    checked="false"
                    classname="RegisterDialog-termsCheckbox-63"
                    color="primary"
                  />
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-11"
                >
                  <mock-box>
                    <mock-typography
                      classname="RegisterDialog-formText-61"
                      color="textSecondary"
                      variant="subtitle1"
                    >
                      I have read the
                       
                      <a
                        class="RegisterDialog-termsLink-64"
                        href="/terms"
                      >
                        Terms and Conditions
                      </a>
                    </mock-typography>
                  </mock-box>
                </div>
              </div>
              <div
                class="MuiGrid-root RegisterDialog-button-62 MuiGrid-item MuiGrid-grid-xs-12"
              >
                <mock-button
                  color="primary"
                  disabled="true"
                  fullwidth="true"
                  size="large"
                  type="submit"
                  variant="contained"
                >
                  CREATE ACCOUNT
                </mock-button>
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
              >
                <mock-typography
                  classname="RegisterDialog-formText-61"
                  color="textSecondary"
                  variant="subtitle1"
                >
                  Have an account?
                   
                  <mock-button
                    color="primary"
                  >
                    SIGN IN
                  </mock-button>
                </mock-typography>
              </div>
            </form>
          </div>
        </div>
      </mock-cardcontent>
    </mock-card>
  </mock-dialog>
  <mock-dialog
    maxwidth="xs"
    open="false"
  >
    <mock-card
      elevation="0"
    >
      <div
        class="MuiCardHeader-root SignInDialog-dialogHeader-66"
      >
        <div
          class="MuiCardHeader-content"
        >
          <mock-typography
            classname="MuiCardHeader-title"
            component="span"
            display="block"
            variant="h5"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <div
                  class="MuiGrid-root MuiGrid-container"
                >
                  <mock-typography
                    variant="h4"
                  >
                    Aqua
                  </mock-typography>
                  <mock-typography
                    classname="SignInDialog-dialogHeaderSecondPart-67"
                    variant="h4"
                  >
                    link
                  </mock-typography>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <mock-iconbutton
                  classname="SignInDialog-closeButton-65"
                  size="small"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </mock-iconbutton>
              </div>
            </div>
          </mock-typography>
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-hidden"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              class="MuiPaper-root MuiAlert-root MuiAlert-standardSuccess MuiPaper-elevation0"
              role="alert"
            >
              <div
                class="MuiAlert-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
                  />
                </svg>
              </div>
              <div
                class="MuiAlert-message"
              >
                Password reset email sent to .
              </div>
              <div
                class="MuiAlert-action"
              >
                <mock-iconbutton
                  aria-label="close"
                  color="inherit"
                  size="small"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </mock-iconbutton>
              </div>
            </div>
          </div>
        </div>
      </div>
      <mock-cardcontent>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
        >
          <div
            class="MuiGrid-root SignInDialog-dialogContentTitle-68 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-10"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <mock-typography
                color="textSecondary"
                variant="h5"
              >
                Sign In
              </mock-typography>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-10"
          >
            <form
              class="SignInDialog-form-69"
            >
              <div
                class="MuiGrid-root SignInDialog-textFieldWrapper-70 MuiGrid-item MuiGrid-grid-xs-12"
              >
                <mock-textfield
                  error="false"
                  fullwidth="true"
                  helpertext=""
                  id="emailAddress"
                  inputprops="[object Object]"
                  label="Email Address"
                  name="emailAddress"
                  placeholder="Email Address"
                  variant="outlined"
                />
              </div>
              <div
                class="MuiGrid-root SignInDialog-textFieldWrapper-70 MuiGrid-item MuiGrid-grid-xs-12"
              >
                <mock-textfield
                  error="false"
                  fullwidth="true"
                  helpertext=""
                  id="password"
                  inputprops="[object Object]"
                  label="Password"
                  name="password"
                  placeholder="Password"
                  type="password"
                  variant="outlined"
                />
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
              >
                <mock-button
                  classname="SignInDialog-forgotPasswordButton-74"
                >
                  <mock-typography
                    color="textSecondary"
                    variant="subtitle2"
                  >
                    Forgot your password?
                  </mock-typography>
                </mock-button>
              </div>
              <div
                class="MuiGrid-root SignInDialog-button-73 MuiGrid-item MuiGrid-grid-xs-12"
              >
                <mock-button
                  color="primary"
                  fullwidth="true"
                  size="large"
                  type="submit"
                  variant="contained"
                >
                  SIGN IN
                </mock-button>
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
              >
                <mock-typography
                  classname="SignInDialog-formText-72"
                  color="textSecondary"
                  variant="subtitle1"
                >
                  Don't have an account?
                   
                  <mock-button
                    color="primary"
                  >
                    SIGN UP
                  </mock-button>
                </mock-typography>
              </div>
            </form>
          </div>
        </div>
      </mock-cardcontent>
    </mock-card>
  </mock-dialog>
  <div
    class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
    role="progressbar"
  >
    <div
      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
    />
    <div
      class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
    />
  </div>
</div>
`;

exports[`Site Detail Page should render with given state from Redux store: snapshot-with-no-data 1`] = `undefined`;
